src/pages/index.js

```diff
@@ -30,17 +34,69 @@ const enableValidation = (settings) => {
 };
 
 function handleImageFormSubmit(name, link) {
-  renderCard({ name: name, link: link });
+  addImagePopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+    .then((newCardData) => {
+      const { name, link } = newCardData;
+      cardSection.addItem(
+        new Card(
+          newCardData,
+          "#card-template",
+          imagePopup,
+          (title, link) => imagePopup.open(title, link),
+          (cardElement, cardID) => {
+            deleteImagePopup.open(cardElement, newCardData._id);
+          },
+          api
+        ).getView()
+      );
+    })
+    .catch((err) => console.error("Error adding new card:", err));
   addImagePopup.close();
 }
 
 function handleProfileFormSubmit(name, description) {
-  userInfo.setUserInfo({
-    name: name,
-    job: description,
-  });
+  editProfilePopup.renderLoading(true);
+  api
+    .updateUserInfo({ name: name, about: description })
+    .then((updatedUser) => {
+      userInfo.setUserInfo({
+        name: updatedUser.name,
+        job: updatedUser.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    });
+}
 
-  editProfilePopup.close();
+function handleAvatarSubmit(link) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateAvatar(link)
+    .then((newAvatar) => {
+      avatarIMG.src = newAvatar.avatar;
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    });
+}
+
+function handleDeleteConfirmation(cardElement, cardID) {
+  deleteImagePopup.renderLoading(true);
+  api
+    .deleteCard(cardID)
+    .then(() => {
+      cardElement.remove();
+      cardElement = null;
+      deleteImagePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
```

Comment:
* When I submit the form for the 1st time the submit button gets `Saving...` for always. You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.